tests/bwc/test_upgrade.py

```diff
@@ -446,3 +449,87 @@ def start_cluster_and_alter_tables(self, version_def, nodes):
                 ALTER TABLE p1 SET (number_of_replicas=1)
             ''')
         self._process_on_stop()
+
+
+class SnapshotCompatibilityTest(NodeProvider, unittest.TestCase):
+
+    CREATE_REPOSITORY = '''
+        CREATE REPOSITORY r1 TYPE S3
+        WITH (access_key = 'minio', 
+              secret_key = 'miniostorage', 
+              bucket='backups', 
+              endpoint = '127.0.0.1:9000', 
+              protocol = 'http')
+        '''
+
+    CREATE_SNAPSHOT_TPT = "CREATE SNAPSHOT r1.s{} ALL WITH (wait_for_completion = true)"
+
+    RESTORE_SNAPSHOT_TPT = "RESTORE SNAPSHOT r1.s{} ALL WITH (wait_for_completion = true)"
+
+    DROP_DOC_TABLE = 'DROP TABLE t1'
+
+    SUPPORTED_VERSION_TRANSITIONS = (
+        ('3.3.x', 5, 5),
```

Comment:
What's the idea behind `idx_start` and `idx_end` ? Why is this necessary?

Reply:
I thought that the version path could go between 3.3.idx_start .. 3.3.idx_end, by making both equal I just take the final revision, but there is always the possibility to run incrementally.